### PR TITLE
Accept a fixed partition size for reset partition in autoinstall.yaml

### PR DESCRIPTION
**Pull request by medicalwei**
This changeset will allow reset_partition to accept int or human-readable disk size (e.g. "12G"), so that the size of reset partition can be fixed.  This will be used for some OEM deployments and automated testing where overwriting or updating reset partition is necessary.

### CODE REVIEW

**Comment by medicalwei** [comment]
There's some underlying serialization issues since attempting to union primitives upsets serialize.py as it seems only taking unions of non-primitive types. Is this intentional?

File: subiquity/server/controllers/filesystem.py [hunk 1]
```diff
@@ -1405,6 +1409,20 @@ async def run_autoinstall_guided(self, layout):
...
+        reset_partition = False
+        reset_partition_size = None
+        rp_input = layout.get("reset-partition", None)
+        if rp_input:
+            reset_partition = True
+            if type(rp_input) in (str, int):
+                reset_partition_size = int(human2bytes(rp_input))
+                log.info(
+                    "autoinstall: will install reset partition"
+                    f"of size {reset_partition_size}"
```
Comment (ogayot) [inline subiquity/server/controllers/filesystem.py:1421]: looks like we're missing a space :)

File: subiquity/server/controllers/tests/test_filesystem.py [hunk 2]
```diff
@@ -459,6 +459,26 @@ async def test_guided_reset_partition(self):
...
+            GuidedChoiceV2(
+                target=target,
+                capability=GuidedCapability.DIRECT,
+                reset_partition=True,
+                reset_partition_size=fixed_reset_size,
+            )
+        )
+        [d1p1, d1p2, d1p3] = self.d1.partitions()
+        self.assertEqual("/boot/efi", d1p1.mount)
+        self.assertEqual(None, d1p2.mount)
```
Comment (ogayot) [inline subiquity/server/controllers/tests/test_filesystem.py:478]:
```python
self.assertIsNone(d1p2.mount)
```

File: subiquity/server/controllers/filesystem.py [hunk 3]
```diff
@@ -1405,6 +1409,20 @@ async def run_autoinstall_guided(self, layout):
                 disk_id=gap.device.id, gap=gap, allowed=[]
             )
 
+        reset_partition = False
+        reset_partition_size = None
+        rp_input = layout.get("reset-partition", None)
+        if rp_input:
+            reset_partition = True
+            if type(rp_input) in (str, int):
```
Comment (ogayot) [inline subiquity/server/controllers/filesystem.py:1417]:
```python
if isinstance(rp_input, (str, int)):
```

**Review by ogayot — APPROVED** [review]
Thanks, LGTM! Just a few minor tweaks

File: subiquity/server/controllers/filesystem.py (hunk 1 above)
Comment (medicalwei) [reply subiquity/server/controllers/filesystem.py:1426]: Added a space in the end of the previous line, thanks!

**Review by dbungert — COMMENTED** [review]
I would like to see some doc/reference/autoinstall-reference.rst material for this, but then I see now that reset-partition isn't documented yet.  Unfortunate.  @medicalwei would you add some content for this in the reference and ignore the bit where the reset_partition bool isn't documented yet?

**Comment by medicalwei** [comment]
@dbungert Reference added, and I hope my writing style doesn't look strange

**Review by dbungert — APPROVED** [review]

**Pull request merged**